### PR TITLE
SIMS #44 Student update

### PR DESCRIPTION
**Pull request by andrewsignori-aot**
Reused the Student Profile page being used for the Student creation to also edit the same.
Added API endpoint to retrieve and update the student contact information.

Added toast notifications for creating an update. After a successful create or update the user will be redirected to the home page.

image

### CODE REVIEW

**Review by fwpushan — APPROVED** [review]

File: sources/packages/web/src/views/Student.vue [hunk 1]
```diff
@@ -169,41 +170,74 @@ interface ProfileState {
...
+    const { handleSubmit, isSubmitting, setValues } = useForm<ProfileState>();
 
-    const onSubmit = handleSubmit(async (formValues) => {
-      const result = await StudentService.shared.createStudent({
-        ...formValues,
-      });
+    onMounted(async () => {
+      if (props.edit) {
+        const contact = await StudentService.shared.getContact();
+        console.log(contact);
```
Comment (2011sandeepraj) [inline sources/packages/web/src/views/Student.vue:193]: Add a comment  on why to use onMounted .Remove console. log

File: sources/packages/web/src/views/Student.vue [hunk 2]
```diff
@@ -169,41 +170,74 @@ interface ProfileState {
...
-        setTimeout(() => {
-          router.push({
-            name: "Home"
-          });
-        }, 1000);
+    const onSubmit = handleSubmit(async formValues => {
+      let redirectHome = true;
+      if (props.edit) {
+        await StudentService.shared.updateStudent({ ...formValues });
+        toast.add({
```
Comment (2011sandeepraj) [inline sources/packages/web/src/views/Student.vue:202]: Create a tech debt for now for Toast Component

File: sources/packages/web/src/views/Student.vue [hunk 3]
```diff
@@ -169,41 +170,74 @@ interface ProfileState {
...
+          summary: "Student Updated",
+          detail: "Student contact information updated!",
+          life: 3000
+        });
       } else {
-        alert(`${result}`);
+        const result = await StudentService.shared.createStudent({
+          ...formValues
+        });
+        if (result === true) {
```
Comment (2011sandeepraj) [inline sources/packages/web/src/views/Student.vue:212]: just mention result can be a boolean or string.

**Review by 2011sandeepraj — APPROVED** [review]
Approving in anticipation. Please run lint as well.

**Pull request merged**